udf/addon_scalars_row.sql

```diff
@@ -0,0 +1,8 @@
+/*
+TODO
+*/
+CREATE TEMP FUNCTION udf_addon_scalars_row(payload ANY TYPE, additional_properties STRING) AS (
+  STRUCT(
+    null as _
```

Comment:
@washort after that you could try filling in this function. i'm hoping addon scalars don't need transformation, but if they do hold off on that part for now and just handle extracting the input columns